assert sock_addr == self.exit_sockets[circuit_id].sock_addr

In latest next branch
commit a717dc8ce2a3c36c2a9b6f0cd28478976fc9d14d
```
ERROR   1420611053.19       community:618   Could not remove relay 1544152986 no activity
ERROR   1420611055.31             log:443   Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/Core/Utilities/twisted_thread.py", line 21, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1192, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1201, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 797, in runUntilCurrent
    f(*a, **kw)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/community/tunnel/community.py", line 1018, in on_data
    self.exit_data(circuit_id, sock_addr, destination, data)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/community/tunnel/community.py", line 1108, in exit_data
    assert sock_addr == self.exit_sockets[circuit_id].sock_addr
exceptions.AssertionError: 

Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/Core/Utilities/twisted_thread.py", line 21, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1192, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1201, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 797, in runUntilCurrent
    f(*a, **kw)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/community/tunnel/community.py", line 1018, in on_data
    self.exit_data(circuit_id, sock_addr, destination, data)
  File "/home/pouwelse/GIT-pulls/tribler/Tribler/community/tunnel/community.py", line 1108, in exit_data
    assert sock_addr == self.exit_sockets[circuit_id].sock_addr
exceptions.AssertionError: 
ERROR   1420611055.31             log:443   Unhandled Error
```

Got it here: http://jenkins.tribler.org/job/Experiment-hiddenservices-seeder-rjruigrok/81/artifact/output/localhost/00004.err/*view*/

Looks like the port of my machine on ubuntu vm on eduroam changes the port very often, causing this assert to fail. 

```
ERROR   1435578999.09             log:143   Unhandled Error
Traceback (most recent call last):
  File "/home/jenkins/workspace/Experiment-hiddenservices-seeder-rjruigrok/gumby/experiments/dispersy/hiddenservices_client.py", line 271, in <module>
    main(HiddenServicesClient)
  File "/home/jenkins/workspace/Experiment-hiddenservices-seeder-rjruigrok/gumby/gumby/experiments/dispersyclient.py", line 516, in main
    reactor.run()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1192, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1201, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 797, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/Experiment-hiddenservices-seeder-rjruigrok/tribler/Tribler/community/tunnel/tunnel_community.py", line 1081, in on_data
    self.exit_data(circuit_id, sock_addr, destination, data)
  File "/home/jenkins/workspace/Experiment-hiddenservices-seeder-rjruigrok/tribler/Tribler/community/tunnel/tunnel_community.py", line 1179, in exit_data
    str(sock_addr), str(self.exit_sockets[circuit_id].sock_addr))
exceptions.AssertionError: ('145.94.155.23', 60030) != ('145.94.155.23', 62350)
```

--- a/Tribler/community/tunnel/tunnel_community.py
+++ b/Tribler/community/tunnel/tunnel_community.py
@@ -1188,10 +1188,11 @@ class TunnelCommunity(Community):
             self.tunnel_logger.error("Dropping data packets, refusing to be an exit node for data")
         elif circuit_id in self.exit_sockets:
             if not self.exit_sockets[circuit_id].enabled:
-                # We got the correct circuit_id, but from a wrong IP.
-                assert sock_addr == self.exit_sockets[circuit_id].sock_addr, "%s != %s" % (
-                    str(sock_addr), str(self.exit_sockets[circuit_id].sock_addr))
-                self.exit_sockets[circuit_id].enable()
+                # Check that we got the correct circuit_id, but from a wrong IP.
+                if sock_addr[0] == self.exit_sockets[circuit_id].sock_addr[0]:
+                    self.exit_sockets[circuit_id].enable()
+                else:
+                    self._logger.error("Dropping outbound relayed packet: IP's are %s != %s", str(sock_addr), str(self.exit_sockets[circuit_id].sock_addr))
             try:
                 self.exit_sockets[circuit_id].sendto(data, destination)
             except:

